rewrite-java/src/main/java/org/openrewrite/java/RenameVariable.java

```diff
@@ -38,35 +43,91 @@ public RenameVariable(J.VariableDeclarations.NamedVariable variable, String toNa
 
     private class RenameVariableByCursor extends JavaIsoVisitor<P> {
         private final Cursor scope;
-        private boolean isConstant = false;
+        private final Stack<Tree> currentNameScope = new Stack<>();
+
+        @Nullable private Tree currentBlock;
```

Comment:
I think it's a little strange to be keeping track of the currentBlock.

Reply:
Jon mentioned postVisit, but I'm not sure how to use it yet. It's possible that'll fix what I was using the currentBlock for.

Try => Try.Resource => single statement => Try.Catch would cause the scope change to be missed.

FileDescriptor fd;
J.Try (J.Try.Resource `J.VariableDeclaration` = J.NewClass) {
   // fd = fs.getFD();
} J.Try.Catch (Exception ex) {
  throw new RuntimeException(targetVal); // would not be renamed.
}